Determinate progress bar

EngrTaofeek opened the pull request:
Created determinate progress bar to show progress based on percentage of convergence. Also added partially transparent dark background to make the progress bar more visible while tracking. Then added tracking instructions indicating phone has to be still while tracking.
One minor challenge(would be highlighted in review) to be fixed before it's ready to be merged.

EngrTaofeek commented on app/src/main/java/org/greenstand/android/TreeTracker/capture/TreeCaptureViewModel.kt:
```diff
@@ -45,8 +45,7 @@ class TreeCaptureViewModel(
     suspend fun captureLocation() {
-        _state.value = _state.value?.copy(isGettingLocation = true)
-//        _state.value = _state.value?.copy(convergencePercentage = treeCapturer.getConvergencePercentage())
+        _state.value = _state.value?.copy(isGettingLocation = true, convergencePercentage = locationDataCapturer.percentageConvergence)
```
@Elforama @ZavenArra this is the only issue left with this PR. The locationDataCapturer.percentageConvergence returns only the default value and doesn't update the state when tracking. But from the logs I can see the value been updated. Any tip on how to make this work

Elforama replied:
Where are you putting the logs that it gets updated.

Elforama replied:
You're just reading a float, to get the value updated you need to listen to it. You'll need to add a convergence listener to the class that updates the value, then you can listen to the value to update the state.

EngrTaofeek commented on app/src/main/java/org/greenstand/android/TreeTracker/models/location/LocationDataCapturer.kt:
```diff
@@ -75,12 +75,12 @@ class LocationDataCapturer(
-//                        )
+                        val minimumConvergenceRatio = min(locationDataConfig.latStdDevThreshold.div(latStdDev).toFloat(),locationDataConfig.lonStdDevThreshold.div(longStdDev).toFloat())
+                        percentageConvergence = min(1f,minimumConvergenceRatio)
+                        Timber.d(
```
@Elforama this is where I logged the value which was been updated as the capture converges

EngrTaofeek commented on app/src/main/java/org/greenstand/android/TreeTracker/models/location/LocationDataCapturer.kt:
```diff
@@ -36,8 +37,7 @@ class LocationDataCapturer(
     private var areLocationUpdatesOn: Boolean = false
-    var percentageConvergence: Float? = null
-        private set
+    var percentageConvergence : Float = 0f
```
@Elforama this

EngrTaofeek commented:
![](https://user-images.githubusercontent.com/38661872/165353384-58f6deb4-7a32-4096-8c1e-d1b6cd77844a.jpeg)
![](https://user-images.githubusercontent.com/38661872/165353396-f3a3b238-5c24-4c8c-b93d-6abeb004a177.jpeg)
@ZavenArra @sebastiangaertner this what this PR does, it adds a dark background to the progress bar and shows the percentage converged along with instruction once the capture button is clicked.

Elforama commented:
A few bit of feedback on this:
- The progress bar and progress are not aligned
- We don't need the decimal place for the percentage
- Do we want to be blocking the users view as they try taking a picture? Perhaps another type of UI would work better, such as a progress line that goes around the edge of the screen?

Elforama commented on app/src/main/java/org/greenstand/android/TreeTracker/models/TreeCapturer.kt:
```diff
@@ -5,6 +5,7 @@ import org.greenstand.android.TreeTracker.models.location.LocationDataCapturer
 import java.io.File
 import java.util.UUID
 import org.greenstand.android.TreeTracker.usecases.CreateTreeUseCase
+import kotlin.math.min
```
No need for this

EngrTaofeek commented:
> A few bit of feedback on this:
> 
> * The progress bar and progress are not aligned
> * We don't need the decimal place for the percentage
> * Do we want to be blocking the users view as they try taking a picture? Perhaps another type of UI would work better, such as a progress line that goes around the edge of the screen?

@Elforama I addressed the first and second in a new commit. For the third, there were complaints that the spinning progress bar was not visible enough which is why I added a dark background to make it more visible under sun light. There were also complaints farmers do not wait for tree convergence to complete before leaving the front of the tree being captured which results into saving random photos that's why the instruction is always there in the dialog while converging.

EngrTaofeek commented:
fixes #970

ZavenArra commented:
<img width="301" alt="image" src="https://user-images.githubusercontent.com/364622/168862989-bc54c784-5861-4a11-a2f3-561b06355c97.png">

Let's make this one final fix, and then I think this PR is good to merge.

EngrTaofeek commented:
updated fix 
![image](https://user-images.githubusercontent.com/38661872/170071187-a8d6dc91-4cab-4562-a8b1-fae238455294.png)

ZavenArra reviewed (approved):
Sending this to QC
